MigFiles/SupportLibraries/ZWaveLib/Controller.cs

```diff
@@ -443,13 +455,21 @@ private void ZwaveMessageReceived(object sender, ZWaveMessageReceivedEventArgs a
                             var znode = devices.Find(n => n.Id == sourceNodeId);
                             if (znode != null)
                             {
-                                byte[] nodeInfo = new byte[nifLength - 2];
+                                // we don't need to exclude the last 2 CommandClasses
+                                byte[] nodeInfo = new byte[nifLength];
                                 //Console.WriteLine(ByteArrayToString(args.Message));
-                                Array.Copy(args.Message, 7, nodeInfo, 0, nifLength - 2);
+                                Array.Copy(args.Message, 7, nodeInfo, 0, nifLength);
                                 znode.NodeInformationFrame = nodeInfo;
+                                var triggerEvents = znode.HandleNodeUpdate(args.Message);
                                 //
-                                RaiseUpdateParameterEvent(new ZWaveEvent(znode, EventParameter.NodeInfo, Utility.ByteArrayToString(nodeInfo), 0));
-                                RaiseUpdateParameterEvent(new ZWaveEvent(znode, EventParameter.WakeUpNotify, "1", 0));
+                                if (!triggerEvents) { 
```

Comment:
I think the name is incorrect HandleNodeUpdate returns true if a this security CommandClass is supported